event_manager: Remove Orphaned sections (#6517)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

  * Added Kconfig options that allow erasing dongle bond on the gaming mouse using buttons or config channel.
  * Added two states to enable erasing dongle peer: ``STATE_DONGLE_ERASE_PEER`` and ``STATE_DONGLE_ERASE_ADV``.

> * Removed:

Reviewer: ```suggestion
```
Should move to updated section as we are not completely removing the kconfig option.

>   * Config channel had been updated and does no longer use orphaned sections to keep information about module id.

Reviewer: ```suggestion
```

    You can remove :kconfig:`CONFIG_LINKER_ORPHAN_SECTION_PLACE` from config file.

* Updated:

> 

Reviewer: ```suggestion

* Config channel no longer uses orphaned sections to store module Id information.
 Hence, the :kconfig:`CONFIG_LINKER_ORPHAN_SECTION_PLACE` option is no longer required in the config file.
```

>    * Documentation and diagrams for the Bluetooth LE bond internal module.

Reviewer: ```suggestion
   * Documentation and diagrams for the :ref:`nrf_desktop_ble_bond` module.
```

Event manager
-------------

> * Removed:

Reviewer: ```suggestion
```
Should move to updated section as we are not completely removing the kconfig option.

>   * Event manager had been updated and does no longer use orphaned sections to keep information about event types, listeners and subscribers.

Reviewer: ```suggestion
```

    You can remove :kconfig:`CONFIG_LINKER_ORPHAN_SECTION_PLACE` from config file.

* Updated:

  * Modified the sections used by the event manager. Stopped using orphaned sections. Removed forced alignment for x86. Reworked priorities.

> 

Reviewer: ```suggestion
* Event manager no longer uses orphaned sections to store information about event types, listeners and subscribers.
Hence, the :kconfig:`CONFIG_LINKER_ORPHAN_SECTION_PLACE` option is no longer required in the config file.
```

Event manager Profiler Tracer
-----------------------------

> * Removed:

Reviewer: ```suggestion
* Updated:
```
Should move to updated section as we are not completely removing the kconfig option.

>   * Event manager Profiler Tracer had been updated and does no longer use orphaned sections to keep profiler information.

Reviewer: ```suggestion
  * Event manager Profiler Tracer no longer use orphaned sections to store profiler information. 
  Hence, the :kconfig:`CONFIG_LINKER_ORPHAN_SECTION_PLACE` option is no longer required in the config file.
```
